pkg/miscellaneous/miscellaneous.go

```diff
@@ -349,3 +355,52 @@ func SleepWithContext(ctx context.Context, sleepFor time.Duration) error {
 		}
 	}
 }
+
+// IsBackupBucketEmpty checks whether the backup bucket is empty or not.
+func IsBackupBucketEmpty(snapStoreConfig *brtypes.SnapstoreConfig, logger *logrus.Logger) bool {
+	logger.Info("Checking whether the backup bucket is empty or not...")
+	store, err := snapstore.GetSnapstore(snapStoreConfig)
+	if err != nil {
+		logger.Fatalf("failed to create snapstore from configured storage provider: %v", err)
+	}
+
+	baseSnap, deltaSnapList, err := GetLatestFullSnapshotAndDeltaSnapList(store)
+	if err != nil {
+		logger.Errorf("failed to list the snapshot: %v", err)
+		return false
+	}
+
+	if baseSnap == nil && (deltaSnapList == nil || len(deltaSnapList) == 0) {
+		logger.Infof("No snapshot found. BackupBucket is empty")
+		return true
+	}
+	return false
+}
+
+// GetInitialClusterState returns the cluster state
+func GetInitialClusterState(ctx context.Context, logger logrus.Entry, podName string, podNS string) string {
```

Comment:
We should pass the Kubernetes client to this function instead of re-creating it every time it is called. The creation and usage of a new client is expensive (mainly due to discovery calls), i.e. users of this function should create a client before and re-use it wherever it is possible. 

Please also add unit tests.

Reply:
> The creation and usage of a new client is expensive (mainly due to discovery calls), i.e. users of this function should create a client before and re-use it wherever it is possible.

yes, agree with you but to address this issue is out of scope of this PR, we can pick this up as separate issue.
Here I can't do much, we have to create new client as `GetInitialClusterState()` is only called once.

Reply:
Hm, I don't think there is much change required.
We just have to amend the function to accept a Kubernetes client and create this client when the BackupRestore server is created. This will also allow you to write unit tests for this function with a fake or mock client.

Reply:
done. PTAL